data/transactions/logic/eval.go

```diff
@@ -1700,6 +1700,36 @@ func opDig(cx *evalContext) {
 	cx.stack = append(cx.stack, sv)
 }
 
+func opCover(cx *evalContext) {
+	depth := int(uint(cx.program[cx.pc+1]))
```

Comment:
Why use multiple type casts?

Reply:
copied this from dig, but I'll dig into it

Reply:
seems like it creates a type mismatch error if its not casted back to an int. without uint it becomes negative

Reply:
Aren't uint64 immediates required to be non-negative?